packages/sdk/src/modules/rbac/index.ts

```diff
@@ -92,4 +518,783 @@ export class RBAC extends Base {
+        throw new Error('Role Id should be 32 char only');
+      if (groupId && groupId.length !== 32)
+        throw new Error('Group Id should be 32 char only');
+      const api = this._getApi();
+      const keyPair = this._metadata?.pair || this._getKeyPair(seed);
+      const unassignRoleToGroupExtrinsics = api.tx?.['peaqRbac']?.[
+        'unassignPermissionToRole'
+      ](stringToU8a(roleId), stringToU8a(groupId));
+      const nonce = await this._getNonce(address || keyPair.address);
+      await this._newSignTx({
+        nonce,
+        address: keyPair,
+        extrinsics: unassignRoleToGroupExtrinsics,
+      });
+      return {
+        hash: unassignRoleToGroupExtrinsics.hash as unknown as CodecHash,
+      };
+    } catch (error) {
+      throw new Error(`Error occurred while unassign role to group: ${error}`);
+    }
+  }
+
+  /**
+   * Unassign role to user.
+   * @param option - The option for unassign role to user.
+   * @returns A promise that resolves when the role is unassign to user.
+   */
+
+  public async unassignRoleToUser(option: UnassignRoleToUser): Promise<
+    | {
+        hash: CodecHash;
+      }
+    | undefined
+  > {
+    try {
+      const { userId, roleId, address = '', seed = '' } = option;
+      if (!roleId) throw new Error('Role Id is required');
+      if (!userId) throw new Error('User Id is required');
+      if (roleId && roleId.length !== 32)
+        throw new Error('Role Id should be 32 char only');
+      if (userId && userId.length !== 32)
+        throw new Error('User Id should be 32 char only');
+      const api = this._getApi();
+      const keyPair = this._metadata?.pair || this._getKeyPair(seed);
+      const unassignRoleToUserExtrinsics = api.tx?.['peaqRbac']?.[
+        'unassignRoleToUser'
+      ](stringToU8a(roleId), stringToU8a(userId));
+      const nonce = await this._getNonce(address || keyPair.address);
+      await this._newSignTx({
+        nonce,
+        address: keyPair,
+        extrinsics: unassignRoleToUserExtrinsics,
+      });
+      return {
+        hash: unassignRoleToUserExtrinsics.hash as unknown as CodecHash,
+      };
+    } catch (error) {
+      throw new Error(`Error occurred while unassign role to user: ${error}`);
+    }
+  }
+
+  /**
+   * Unassign user to group.
+   * @param option - The option for unassign user to group.
+   * @returns A promise that resolves when the user is unassign to group.
+   */
+
+  public async unassignUserToGroup(option: UnassignUserToGroup): Promise<
+    | {
+        hash: CodecHash;
+      }
+    | undefined
+  > {
+    try {
+      const { userId, groupId, address = '', seed = '' } = option;
+      if (!groupId) throw new Error('Group Id is required');
+      if (!userId) throw new Error('User Id is required');
+      if (groupId && groupId.length !== 32)
+        throw new Error('Group Id should be 32 char only');
+      if (userId && userId.length !== 32)
+        throw new Error('User Id should be 32 char only');
```

Comment:
Too many repetition of these lines of code in different functions.

Create a validation function and call it instead.

Reply:
make a validation function